src/interface/meshblock_data.hpp

```diff
@@ -155,23 +157,51 @@ class MeshBlockData {
 
     // special case when the list of vars is empty, copy everything
     if (vars.empty()) {
-      for (auto v : src->GetVariableVector()) {
-        add_var(v);
+      if constexpr (std::is_same_v<SRC_t, MeshBlockData<Real>>) {
+        for (auto v : src->GetVariableVector()) {
+          add_var(v);
+        }
+      } else if constexpr (std::is_same_v<SRC_t, MeshBlock>) {
+        for (auto const &q : resolved_packages->AllFields()) {
+          AddField(q.first.base_name, q.second, q.first.sparse_id);
+        }
       }
     } else {
-      for (const auto &v : vars) {
-        auto var = src->GetVarPtr(v);
-        add_var(var);
-        // Add the associated flux as well if not explicitly
-        // asked for
-        if (var->IsSet(Metadata::WithFluxes)) {
-          auto flx_name = var->metadata().GetFluxName();
-          bool found = false;
-          for (const auto &v2 : vars) {
-            if (src->GetVarPtr(v2)->label() == flx_name) found = true;
+      if constexpr (std::is_same_v<SRC_t, MeshBlockData<Real>>) {
+        for (const auto &v : vars) {
+          auto var = src->GetVarPtr(v);
+          add_var(var);
+          // Add the associated flux as well if not explicitly
+          // asked for
+          if (var->IsSet(Metadata::WithFluxes)) {
+            auto flx_name = var->metadata().GetFluxName();
+            bool found = false;
+            for (const auto &v2 : vars) {
+              if (src->GetVarPtr(v2)->label() == flx_name) found = true;
+            }
+            if (!found) add_var(src->GetVarPtr(flx_name));
           }
-          if (!found) add_var(src->GetVarPtr(flx_name));
         }
+      } else {
+        PARTHENON_FAIL(
+            "Variable subset selection not yet implemented for MeshBlock input.");
+      }
+    }
+
+    // TODO(LFR): Not sure why we only do this in the MeshBlock case, but this carries
+    // over from the previous iteration.
+    if constexpr (std::is_same_v<SRC_t, MeshBlock>) {
+      if (stage_name_ == "base") {
+        const auto &swarm_container = GetSwarmData();
+        swarm_container->Initialize(resolved_packages, GetBlockSharedPointer());
+      }
```

Comment:
Ping @pdmullen @brryan

Reply:
I don't see any issue with this. Swarms are `OneCopy`-like and always belong to the base stage so it doesn't really make sense to `Add` a new `MeshBlockData` to the `DataCollection` based on an existing `MeshBlockData`. We have thought a bit about the best way to express this through the parthenon interface but don't have any good thoughts at the moment.

Reply:
What was the concern here? Sorry I am late to the party...

Reply:
I just wanted to know if this was the desired behavior and if there were any side effects of what I did.

Reply:
I'll test downstream first thing tomorrow.

Reply:
Did the test happen? @pdmullen

Reply:
They did, there is an old comment from Patrick below.